pkgs/development/python-modules/fairseq/default.nix

```
@@ -0,0 +1,88 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+
+# Native build inputs
+, cython
+, which
+
+# Propagated build inputs
+, cffi
+, hydra-core
+, omegaconf
+, sacrebleu
+, numpy
+, regex
+, torch
+, tqdm
+, bitarray
+, torchaudio
+, scikit-learn
+, packaging
+
+# Check inputs
+, pytestCheckHook
+#, apex
+}:
+let
+  pname = "fairseq";
+  version = "0.12.3";
+in
+buildPythonPackage rec {
+  inherit version pname;
+
+  # Pypi source package doesn't contain tests
+  src = fetchFromGitHub {
+    owner = "pytorch";
+    repo = pname;
+    rev = "v${version}";
+    hash = "";
+  };
+
+  disabled = pythonOlder "3.7";
+
+  nativeBuildInputs = [
+    cython
+    which
+  ];
+
+  propagatedBuildInputs = [
+    cffi
+    hydra-core
+    omegaconf
+    sacrebleu
+    numpy
+    regex
+    torch
+    tqdm
+    bitarray
+    torchaudio
+    scikit-learn
+    packaging
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+    # apex
+  ];
+
+  pythonImportsCheck = [ "fairseq" ];
+
```

```suggestion
  preCheck = ''
    export HOME=$TMPDIR
    cd tests
  '';
  
  pytestFlagsArray = [
    "--import-mode append"
  ];
```
The pytest used in the current `pytestCheckHook` cannot find the C extension module.